compiler/src/main/kotlin/dev/zacsweers/metro/compiler/NameAllocator.kt

```diff
@@ -22,6 +22,44 @@ import kotlin.uuid.Uuid
 import org.jetbrains.kotlin.name.Name
 import org.jetbrains.kotlin.renderer.KeywordStringsGenerated.KEYWORDS
 
+/**
+ * Cross-platform reserved keywords that must be avoided in generated identifiers.
+ *
+ * This set contains reserved keywords from JavaScript (ES5/ES6), C99, and Objective-C
+ * that are NOT already included in Kotlin's KEYWORDS constant. By pre-allocating these
+ * keywords, we ensure generated code compiles successfully on all Kotlin multiplatform
+ * targets (JVM, JS, Native).
+ *
+ * Sources:
+ * - JavaScript: ECMAScript 5.1 spec (https://262.ecma-international.org/5.1/#sec-7.6)
+ * - C99: ISO/IEC 9899:1999 C Standard
+ * - Objective-C: Apple Objective-C Documentation + Clang keywords
+ *
+ * Many keywords overlap across platforms (e.g., `break`, `case`, `for`), so this set
+ * only contains platform-specific keywords not already blocked by Kotlin.
+ */
+private val CROSS_PLATFORM_RESERVED_KEYWORDS = setOf(
+  // JavaScript-specific keywords and special identifiers
+  // (not already in Kotlin KEYWORDS)
+  "arguments", "await", "debugger", "delete", "eval", "function",
+  "in", "instanceof", "let", "typeof", "var", "void", "with", "yield",
+
+  // C99-specific keywords (not already in Kotlin KEYWORDS)
+  "auto", "char", "double", "extern", "float", "goto", "inline", "int",
+  "long", "register", "restrict", "short", "signed", "sizeof", "struct",
+  "typedef", "union", "unsigned", "volatile", "_Bool", "_Complex", "_Imaginary",
+
+  // Objective-C-specific keywords (not already in Kotlin KEYWORDS)
+  "id", "nil", "Nil", "YES", "NO", "SEL", "IMP", "BOOL", "instancetype",
+  "required", "optional", "synthesize", "dynamic", "readonly", "readwrite",
+  "assign", "retain", "copy", "nonatomic", "atomic", "strong", "weak",
+)
+
+/**
+ * Dangerous characters that must be replaced in identifiers.
+ */
+private val DANGEROUS_CHARS = setOf('.', ';', '/', '<', '>', '[', ']')
```

Comment:
Let's make this just a string like

```suggestion
private const val DANGEROUS_CHARS = ".l/<>[]"
```

The `in` check will still work but it's a bit faster